### ISSUE TITLE
Landmark Component

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#3991]
When using facebook.com, I noticed something when using tab to interact with the site, namely a bar that shows at the top of the screen:

![image](https://user-images.githubusercontent.com/3901764/64913434-cae00c80-d705-11e9-88d6-215f12acaec2.png)

When interacting with the menus, it will go through and show focus for each region of the page. It also provides helpers for navigating other parts of the platform, alongside general accessibility guidance on the right-hand side.

Looking towards our UI Shell, I was wondering what we could offer in the same vein in a component I'm calling a landmark component (name TBD).

Some quick explorations included:

**Mobile** 
_Note: this overflows if no space as on mobile folks would be using native VO versus at a computer with a keyboard_

![image](https://user-images.githubusercontent.com/3901764/64913459-31652a80-d706-11e9-8196-d9f2b4470fc3.png)

**Tablet or Desktop**
![image](https://user-images.githubusercontent.com/3901764/64913461-388c3880-d706-11e9-96d3-23a64d4e842d.png)

**Opening menu on desktop with section**
![image](https://user-images.githubusercontent.com/3901764/64913463-4215a080-d706-11e9-9ebe-abaa85094f17.png)

I'd be curious to see if this is something we should explore on the accessibility side cc @dakahn, it seems like it'd help out a ton in terms of framing how one might want to think about structuring their websites such that these kinds of landmarks are obvious/intuitive/and ultimately navigable.

On the design side, I'm curious what you all think of this in terms of the user experience and, if worth pursuing, what changes I could make so that this component matches the broader visual design of Carbon 😄 cc @carbon-design-system/design

**Comment by dakahn** [comment]
This is a fantastic idea! If implemented correctly something like this could improve usability for all keyboard users (people with disabilities and power users etc).

What do they put inside "Accessibility Help"?

**Comment by designertyler** [comment]
This would be great to add as a component and/or as a11y guidance in the Navigation pattern.

**Comment by chrisconnors-ibm** [comment]
how do you invoke this? (oh, `opt-/`). it's specifically a facebook accessibility accommodation.

It inverts the information hierarchy in a very strange way. I'm not even sure what that gets you in terms of accessibility other than clustering focus.

It oddly mixes their context ("Selections on this page") and platform ("Other pages on Facebook") navigations items into nearly identical and physically adjacent embodiments. it duplicates functionality already present.

We shouldn't confound our existing embodiments this way.

**Comment by joshblack** [comment]
@chrisconnors-ibm

>  I'm not even sure what that gets you in terms of accessibility other than clustering focus

I think that's very much so it, it offers a one-stop place to navigate and understand the structure of a page without having to physically interact with it through a keyboard.

> it duplicates functionality already present.

I think this might understate the difference for how a keyboard user interacts with a site. If is difficult for a user to parse the visual information of a page, then it is helpful to enumerate the regions or landmarks on the page without having to visit each one.

If it is a user who is able to parse visual information but is using a keyboard, the act of navigating to a part of the page that offers what they can see is not immediately apparent and this control offers them this behavior that mouse users may take for granted.

For a full demo of the behavior, made a quick gif:

![demo](https://user-images.githubusercontent.com/3901764/64973775-3db9c680-d871-11e9-985c-5af24114d3a2.gif)

@dakahn 

> What do they put inside "Accessibility Help"?

Seems like they offer the following information:

![image](https://user-images.githubusercontent.com/3901764/64974263-14e60100-d872-11e9-9adf-396bfc862be5.png)

Would be cool if we could link to IBMa in the last two links

**Comment by dakahn** [comment]
@chrisconnors-ibm I'm guessing by "it's specifically a facebook accessibility accomodation" you mean that their product's complexity merits a feature like this?

@joshblack said a lot of what I was going to say, but I'll add one more point. This "information hierarchy" as referenced is already subverted entirely by rotors (web and mobile) found in screen readers. User's have the ability to jump to any page landmark, element, link, visited links, table, form input etc.

https://support.apple.com/en-us/HT204783

Often a keyboard (sans screen reader) user's experience on information dense websites (even those totally "keyboard accessible") can be very tedious and frustrating (focus traps, endlessly long navigation menus etc).

Think of accessibility affordances like this as failsafes. Maybe you don't need to quickly jump to the form in question -- maybe they've thoughtfully designed and programmed a "skip to main" button

but probably not and so here's helping hand.

**Comment by laurenmrice** [comment]
not stale?

**Comment by aagonzales** [comment]
Would this be part of the accessibility project? Should we put it in the backlog or is this a discussion thread?

**Comment by joshblack** [comment]
@aagonzales I think it's still discussion right now, no planned efforts or anything. Might require research to suggest that this would indeed be helpful/useful before going for it

**Comment by laurenmrice** [comment]
not stale
